src/number_theory/bernoulli.lean

```diff
@@ -328,3 +329,55 @@ begin
   rw [cast_mul, cast_mul, mul_div_mul_right _ _ hj, add_choose, cast_dvd_char_zero],
   exact factorial_mul_factorial_dvd_factorial_add i j,
 end
+
+
+section faulhaber
+
+/-- Faulhabers' theorem: sum of powers. -/
+theorem faulhaber (n p : ℕ) :
```

Comment:
I know this sounds like a heresy, but even for famous theorems names we tend to follow mathlib's naming convention (and mention the famous name in the docstring, like you do). So this would be `sum_range_pow`, I guess.
The benefit of this approach, is that people can stumble on the theorem via auto-complete, even if they have never heard of Faulhaber before. And people who do know Faulhaber can find it via search for the name and hitting the docstring.

Reply:
I think a more descriptive docstring is also in order.

Reply:
I think the name `sum_range_power` is great. I think people tend rather remember that there exists a formula for the sum of powers then its name.